### ISSUE TITLE
invalid code lengths set exception when uploading a Firefox profile containing a shared library

### ISSUE THREAD

**Issue by fippo** [SeleniumHQ/selenium#1884]
## Meta -
OS: Linux
Selenium Version: 2.53.0 server, selenium-webdriver 2.53.1 javascript bindings as client
Browser: Firefox
Browser Version: 45.0

## Expected Behavior -
A profile containing a shared library is transferred to the selenium server correctly.
This is likely similar to this issue (probably #640) from the selenium-webdriver JS changelog for 2.48.0:
```
* FIXED: zip files created for transfer to a remote WebDriver server are no
    longer compressed. If the zip contained a file that was already compressed,
    the server would return an "invalid code lengths set" error.
```

## Actual Behavior -
the client exits with:
WebDriverError: invalid code lengths set
The server throws a lengthy exception ([full log](https://github.com/SeleniumHQ/selenium/files/190405/selenium-issue.txt)) which boils down to 
```
Caused by: org.openqa.selenium.WebDriverException: java.util.zip.ZipException: invalid code lengths set
```

## Steps to reproduce -
1) run selenium-server-standalone-2.53.0.jar
2) run attached nodejs script [bug.txt](https://github.com/SeleniumHQ/selenium/files/190409/bug.txt) with the contents of [this zip](https://github.com/SeleniumHQ/selenium/files/190410/buggyprofile.zip) unzipped in the proper location

**Comment by fippo** [comment]
the base64 encoded profile that is transferred via http as a zip looks somewhat odd. Attached as
[profile.zip](https://github.com/SeleniumHQ/selenium/files/190492/profile.zip)
The libusrsctp.so is corrupt which leads to errors decompressing this. In that profile.zip unzip -v shows this:
```
 Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
 2172115  Defl:N   829967  62% 2016-03-06 13:31 7633a1c0  libusrsctp.so
```
whereas running unzip -v on the zip of the directory I provided shows this:
```
 Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
--------  ------  ------- ---- ---------- ----- --------  ----
 2172115  Defl:N   840593  61% 2016-03-26 09:54 7633a1c0  buggyprofile/libusrsctp.so
```
Note that the compressed size is different while the length and crc32 are the same.

**Comment by fippo** [comment]
looks like adm-zip is adding the so in a wrong way. Setting the compression method to 0 (no compression) which is current for the first entry [here](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/firefox/profile.js#L388) for all entries 'fixes' this but doesn't compress things at all.

**Comment by fippo** [comment]
@jleyba ping (since you fixed #640)? Happy to provide a PR which doesn't compress all files if that helps! Possibly even a smart variant which looks for characters outside the ascii range before trying to compress but I can't make promises here
